platform/core/src/studies/services/wado/retrieveMetadata.js

```diff
@@ -452,18 +460,101 @@ async function resultDataToStudyMetadata(server, studyInstanceUid, resultData) {
  * @returns {Promise}
  */
 async function RetrieveMetadata(server, studyInstanceUid) {
-  const config = {
+  return (server.enableStudyLazyLoad
+    ? lazyLoadStudyMetadata
+    : loadStudyMetadata)(server, studyInstanceUid);
+}
+
+async function loadStudyMetadata(server, studyInstanceUID) {
+  const dicomWeb = new api.DICOMwebClient({
     url: server.wadoRoot,
     headers: DICOMWeb.getAuthorizationHeader(server),
-  };
-  const dicomWeb = new api.DICOMwebClient(config);
-  const options = {
-    studyInstanceUID: studyInstanceUid,
-  };
+  });
+  return dicomWeb
+    .retrieveStudyMetadata({ studyInstanceUID })
+    .then(result => createStudyFromInstanceList(server, result));
+}
+
+async function lazyLoadStudyMetadata(server, studyInstanceUid) {
+  const seriesIds = await searchStudySeries(server, studyInstanceUid);
+  if (seriesIds) {
+    const dicomWeb = new api.DICOMwebClient({
+      url: server.wadoRoot,
+      headers: DICOMWeb.getAuthorizationHeader(server),
+    });
```

Comment:
Both load methods can use the same client instance. No need to create it in both implementations?

Reply:
one uses `wadoRoot` the other one uses `qidoRoot` (which might be different according to the config)...

Reply:
Thanks for the call-out! I think I need to take a closer look at DICOMWebClient. I assumed the URLs exposed APIs that adhered to different contracts (requests + responses). It's interesting that you can provide it either URL and it works.

I'll do my own research before poking you on this one again ^_^